integration-tests/tests/src/tests/sync/mongo-db-client.ts

```diff
@@ -93,6 +93,438 @@ describe.skipIf(environment.missingServer, "MongoDB Client", function () {
+        await insertThreeDocuments();
+
+        const oldDoc = await collection.findOneAndDelete({ _id: nonExistentId });
+        expect(oldDoc).to.be.null;
+
+        const count = await collection.count();
+        expect(count).to.equal(3);
+      });
+    });
+
+    describe("#insertOne", function () {
+      it("inserts document with id", async function (this: AppContext & Mocha.Context) {
+        const result = await collection.insertOne({ _id: insertedId1 });
+        expect(result.insertedId).to.equal(insertedId1);
+
+        const count = await collection.count();
+        expect(count).to.equal(1);
+      });
+
+      it("inserts document without id", async function (this: AppContext & Mocha.Context) {
+        const result = await collection.insertOne({ text: insertedText });
+        expect(result.insertedId).to.be.instanceOf(BSON.ObjectId);
+
+        const count = await collection.count();
+        expect(count).to.equal(1);
+      });
+
+      it("throws if inserting document with existing id", async function (this: AppContext & Mocha.Context) {
+        await collection.insertOne({ _id: insertedId1 });
+
+        await expect(collection.insertOne({ _id: insertedId1 })).to.be.rejectedWith("Duplicate key error");
+      });
+    });
+
+    describe("#insertMany", function () {
+      it("inserts documents with ids", async function (this: AppContext & Mocha.Context) {
+        const result = await collection.insertMany([{ _id: insertedId1 }, { _id: insertedId2 }, { _id: insertedId3 }]);
+        expect(result.insertedIds).to.have.length(3);
+
+        const count = await collection.count();
+        expect(count).to.equal(3);
+      });
+
+      it("inserts documents without ids", async function (this: AppContext & Mocha.Context) {
+        const result = await collection.insertMany([
+          { text: insertedText },
+          { text: insertedText },
+          { text: insertedText },
+        ]);
+        expect(result.insertedIds).to.have.length(3);
+        for (const insertedId of result.insertedIds) {
+          expect(insertedId).to.be.instanceOf(BSON.ObjectId);
+        }
+
+        const count = await collection.count();
+        expect(count).to.equal(3);
+      });
+
+      it("throws if inserting document with existing id", async function (this: AppContext & Mocha.Context) {
+        await collection.insertMany([{ _id: insertedId1 }]);
+
+        await expect(collection.insertMany([{ _id: insertedId1 }])).to.be.rejectedWith("Duplicate key error");
+      });
+    });
+
+    describe("#updateOne", function () {
+      const updatedText = "Updated text";
+
+      it("updates specific document", async function (this: AppContext & Mocha.Context) {
+        await insertThreeDocuments();
+
+        const result = await collection.updateOne({ _id: insertedId3 }, { $set: { text: updatedText } });
+        expect(result).to.deep.equal({ matchedCount: 1, modifiedCount: 1 });
```

Comment:
one suggestion here could also be to fetch the updated document and validate that it contains the new value for `text`

Reply:
there are other cases in the file where we rely on `expect(result).to.deep.equal(matchedcount..., modifiedCount...)`. I don't know how petty we want to be but I think it would be worth adding those extra checks as they're not too difficult to write nor imply too much overhead.

Reply:
Thanks, I'll add that :) (I'll go ahead and apply that to e.g. deletions as well, checking that the fetched doc is `null`).